cluster/aws/util.sh

```diff
@@ -721,14 +723,11 @@ function create-iam-profile {
 
 # Creates the IAM roles (if they do not already exist)
 function ensure-iam-profiles {
-  aws iam get-instance-profile --instance-profile-name ${IAM_PROFILE_MASTER} || {
```

Comment:
> This is the modus operandi for resource allocation- if we create it automatically, we clean it up automatically as well.

This isn't entirely true. We do take a more conservative approach at several places and that's the most reasonable thing to do. Examples being: Federated Services, Federated Replica Sets, Kubernetes Deployments and in various other places.

Moreover, what makes it reasonable to assume that these resources were created entirely by `kube-up` script? Do we add an annotation or something that we check against before deleting it? Is there a hash that's hard to reproduce?